lib/jekyll/filters.rb

```diff
@@ -13,10 +13,12 @@ module Filters
     # input - The Markdown String to convert.
     #
     # Returns the HTML formatted String.
-    def markdownify(input)
+    def markdownify(input, mode = nil)
       site = @context.registers[:site]
       converter = site.find_converter_instance(Jekyll::Converters::Markdown)
-      converter.convert(input.to_s)
+      output = converter.convert(input.to_s)
+      output = output.gsub(%r!<p>(.*?)<\/p>\n!im, '\1') if mode.to_s == "inline"
```

Comment:
`gsub` is too fragile for something like this. We have discussed better solutions.